### PR TITLE
He mh share list

### PR DESCRIPTION
**Pull request by haleyelder**
The goal of this PR is to add two buttons to the home screen to either generate a new token to create a new list or search the current database to validate with an existing token. Pressing the 'get new token' button will generate the token, save it to `localStorage`, and allow the user to press 'submit' on the page to create a new list with said token.

Pressing the 'existing token' button will show a default view of an input field to either paste or type in an existing token to pull in data from the specified list. Until the token validates, the page will show there is "No List Found" and an 'add first item' button below till the database can validate the token is present. The 'add first item' button will direct users to the Form.js component to add the first item. 

This week, we focused on the functionality in the App.js to add in similar functionality from the Form.js file to add to the Firebase database on token and list creation. We found that submitting the token for list creation, so far, will need to push an empty object within the document, with next list additions to follow after this empty object. 

To toggle the button views, there are state hooks and conditional rendering to show or hide specific inputs depending what is pressed `onClick`.

### CODE REVIEW

**Review by RachaelSkye — APPROVED** [review]
Looks good to me! I wouldn't change anything. I like how you added the navlink to go to add items if the list is empty.

File: src/App.js [hunk 1]
```diff
@@ -26,22 +29,57 @@ function App() {
...
   const enterTokenHandler = e => {
     setToken(e.target.value);
   };
 
+  const existingToken = () => {
+    setShowNew(false);
+    setShowExisting(true);
+  };
+
+  // creates new collection with a blank document
```
Comment (rmorabia) [inline src/App.js:46]: Love to see comments!!!!

File: src/App.js [hunk 2]
```diff
@@ -6,11 +6,14 @@ import Header from "./Header/Header";
...
 import getToken from "./lib/token";
 import Form from "./Form/Form";
+import * as firebase from "./lib/firebase";
 
 function App() {
   const [isListSelected, setIsListSelected] = useState(true);
   const [isAddSelected, setIsAddSelected] = useState(false);
   const [token, setToken] = useState("");
+  const [showNew, setShowNew] = useState(false);
+  const [showExisting, setShowExisting] = useState(false);
```
Comment (rmorabia) [inline src/App.js]: So, I noticed for these two, you're using comments down below -- I think those comments might be extraneous if you named these better. ie. `showNewToken` & `showExistingToken`. Long variable names are fine as long as they are descriptive.

**Review by rmorabia — APPROVED** [review]
Great job, y'all! Just one suggestion, but very readable overall.

**Review by msholty-fd — APPROVED** [review]
I checked out the branch and ran this locally. It works well! 👏 

My only thought is that perhaps we would want the "show new" vs "show existing" states to be separate routes? 🤔 

Then I think we wouldn't have to hold any state to determine what to show - we'd just route the user to that page and allow that page's component to do its thing. What do you think? (I am not suggesting you make this change now, just wanted to discuss it a bit)

File: src/App.js (hunk 2 above)
Comment (mjhcodes) [reply src/App.js]: Thanks for this suggestion. Naming things is hard. Haha this was actually the 2nd name we gave to these. I've just incorporated these changes.

File: src/List/ItemList.js [hunk 3]
```diff
@@ -28,11 +29,20 @@ const Items = props => {
...
+            <button>Add 1st Item</button>
+          </NavLink>
+        </Fragment>
+      ) : (
+        <ul>
+          {dbItems.map((item, index) => {
+            return <li key={index}>{item.name}</li>;
+          })}
+        </ul>
+      )}
```
Comment (jcwesley93) [inline src/List/ItemList.js:45]: I appreciate how clean the logic flow is here!

**Review by jcwesley93 — COMMENTED** [review]
Overall, I think you both did a great job. The code and logic is easy to understand, and I like the small functional changes that were added.

**Review by jcwesley93 — APPROVED** [review]

**Pull request merged**